draft-ietf-tls-tls13.md

```diff
@@ -3524,12 +3524,11 @@ that:
   types. That is, if a handshake message is split over two or more
   records, there MUST NOT be any other records between them.
 
-- Handshake messages MUST NOT span key changes. Because
-  the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate messages
-  can arrive immediately prior to a key change,
-  upon receiving these messages a receiver MUST verify that the end
-  of these messages aligns with a record boundary; if not, then it MUST
-  terminate the connection with an "unexpected_message" alert.
+- Handshake messages MUST NOT span key changes. Implementations MUST verify that
+  all messages immediately preceding a key change align with a record boundary.
+  Because the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate
+  messages can immediately precede a key change, implementations MUST send these
+  messages in alignment with a record boundary.
```

Comment:
This loses the alert text, which I would add after the second sentence: "When applying a key change to remote keys, an implementation that has received a partial handshake message MUST generate a fatal "unexpected_message" alert.  

This also doesn't really fix the problem for a ClientHello that precedes a HelloRetryRequest.  I would change the last sentence here to be advisory (the MUST in the first two sentences suffices), as in "ClientHello, .... and KeyUpdate messages all immediate precede a key change."